crates/rust-analyzer/src/to_proto.rs

```diff
@@ -308,7 +309,12 @@ pub(crate) fn semantic_tokens(
     line_index: &LineIndex,
     highlights: Vec<HighlightedRange>,
 ) -> lsp_types::SemanticTokens {
-    let mut builder = semantic_tokens::SemanticTokensBuilder::default();
+    let id = match SystemTime::now().duration_since(SystemTime::UNIX_EPOCH) {
```

Comment:
Hm, using system time seems iffy. Let's use `static COUNTER: AtomitU32` instead.

Reply:
`SystemTime` can go backwards. If you don't want this, you need to use `Instant`.